k8s/runners/testing/graviton/2/release.yaml

```diff
@@ -0,0 +1,86 @@
+---
+apiVersion: helm.fluxcd.io/v1
+kind: HelmRelease
+metadata:
+  name: runner-testing-graviton2
+  namespace: gitlab
+spec:
+  releaseName: testing-graviton2
+  chart:
+    name: gitlab-runner
+    repository: https://charts.gitlab.io
+    version: 0.37.2  # gitlab-runner@14.7.0
+  values:
+    image: gitlab/gitlab-runner:alpine-v14.7.0
+    imagePullPolicy: IfNotPresent
+    replicas: 1
+
+    gitlabUrl: "https://gitlab.spack.io/"
+    unregisterRunners: true
+    terminationGracePeriodSeconds: 21600  # six hours
+    concurrent: 2
+    checkInterval: 30
+
+    metrics:
+      enabled: true
+
+    rbac:
+      serviceAccountName: runner
+
+    runners:
+      config: |
+        [[runners]]
+          output_limit = 4096
+          environment = ["FF_GITLAB_REGISTRY_HELPER_IMAGE=1"]
+          [runners.kubernetes]
+            helper_image = "gitlab/gitlab-runner-helper:arm-latest"
+            privileged = false
+
+            cpu_request = "750m"
+            cpu_request_overwrite_max_allowed = "16"
+            cpu_limit = "16"
+            cpu_limit_overwrite_max_allowed = "16"
+
+            memory_request = "2G"
+            memory_request_overwrite_max_allowed = "64G"
+            memory_limit = "64G"
+            memory_limit_overwrite_max_allowed = "64G"
+
+            namespace = "pipeline"
+            poll_timeout = 600  # ten minutes
+            service_account = "runner"
+            [runners.kubernetes.pod_annotations]
+              "fluentbit.io/exclude" = "true"
+              "gitlab/ci_pipeline_url" = "$CI_PIPELINE_URL"
+              "gitlab/ci_job_url" = "$CI_JOB_URL"
+              "gitlab/ci_project_url" = "$CI_PROJECT_URL"
+              "gitlab/ci_runner_description" = "$CI_RUNNER_DESCRIPTION"
+              "gitlab/ci_job_id" = "$CI_JOB_ID"
+            [runners.kubernetes.pod_labels]
+              "gitlab/ci_job_size" = "$CI_JOB_SIZE"
+              "metrics/gitlab_ci_pipeline_id" = "$CI_PIPELINE_ID"
+              "metrics/gitlab_ci_project_namespace" = "$CI_PROJECT_NAMESPACE"
+              "metrics/gitlab_ci_project_name" = "$CI_PROJECT_NAME"
+              "metrics/gitlab_ci_job_stage" = "$CI_JOB_STAGE"
+              "metrics/gitlab_ci_commit_ref_name" = "$CI_COMMIT_REF_NAME"
+              "metrics/spack_ci_stack_name" = "$SPACK_CI_STACK_NAME"
+              "metrics/spack_job_spec_pkg_name" = "$SPACK_JOB_SPEC_PKG_NAME"
+              "metrics/spack_spec_needs_rebuild" = "$SPACK_SPEC_NEEDS_REBUILD"
+            [runners.kubernetes.node_selector]
+              "spack.io/node-pool" = "glr-graviton2"
+
+      # default image
+      image: "busybox:1.32.0"
+      imagePullPolicy: "if-not-present"
+      locked: false
+
+      tags: "testing2,arm,aarch64,graviton,graviton2,small,medium,large,huge"
```

Comment:
I'm confused about some of the tags shared by almost all of these runners. Namely:

* `testing2`
* `small`
* `medium`
* `large`
* `huge`

As is, a job could just be tagged with `huge`, and it would potentially run on any of these? Is that desired? If so, feel free to ignore this comment, but I just wanted to make sure of that behavior.

Reply:
Yes, your understanding is correct. A job can only be served by a runner that matches all of its tags. But the inverse is not true. Extra tags on a runner do not prevent it from picking a job it otherwise would be able to run.

In the not-too-distant future, I think we should drop all of the small/medium/large/etc flags from our runners and the jobs that get generated by Spack, but that's a task for another PR.